packages/dds/shared-object-base/src/serializer.ts

```diff
@@ -82,6 +85,8 @@ export type FluidSerializableReadOnly =
 /**
  * Check if a value in {@link FluidSerializableReadOnly} data is an {@link @fluidframework/core-interfaces#IFluidHandle}.
  *
+ * Warning: Non-IFluidHandle objects with a key "IFluidHandle" used to be unconditionally interpreted as IFluidHandle and currently assert.
```

Comment:
Maybe be more explicit here about the "used to" and "currently", something like

```suggestion
 * Warning: Non-IFluidHandle objects with a key "IFluidHandle" will cause an assert. Prior to 2.0.0-internal.7.1, they were unconditionally interpreted as IFluidHandle.
```

Reply:
Done.